Homework25

rogarciap8 opened the pull request.

rogarciap8 committed:
Saving work

rogarciap8 committed:
HW20 Done

rogarciap8 committed:
Homework 21 done

rogarciap8 committed:
Homework22

rogarciap8 committed:
Made elements protected/private

rogarciap8 committed:
Homework 23

rogarciap8 committed:
Homework24

rogarciap8 committed:
Updated code

rogarciap8 committed:
Homework 25

Berichicko commented:
Please, add more test cases in TestNG.xml and run it.

For example 

![image](https://github.com/Test-Pro-Bootcamp/A45-QA-Automation-Framework/assets/104761482/29e883ab-8183-4a82-8671-b335d4e2e137)

HINT

I'll try to help you, here's an example of parallel testing in 4 threads on a local machine:



_Download_

1) Code with Russian comments https://drive.google.com/file/d/1dQjnroe0pIiIqll_dT8rTwgAO5cekj4W/view?usp=share_link


2) Code with English comments https://drive.google.com/file/d/1H6MUQcisY_XQ2kn47AhrGU5TBVxli1ae/view?usp=share_link


**To run the tests, open the README file in IntelliJ IDEA**


![image](https://github.com/Test-Pro-Bootcamp/A45-QA-Automation-Framework/assets/104761482/3c7532eb-65f0-4854-823b-be41ba43622a)

**click**  ![image](https://github.com/Test-Pro-Bootcamp/A45-QA-Automation-Framework/assets/104761482/7313f1bd-6aba-4a26-8f83-dd510a432c8c)

_PS:  To unzip a ZIP file on a Mac_ https://www.winzip.com/en/learn/file-formats/zip/mac/

**Do I have to run tests in Lambdatest or on my local machine?**
_It depends on your needs. Lambdatest is for cross-browser and device testing, but running tests locally is fine_

rogarciap8 commented:
@Berichicko I think you meant to post this on someone else's feedback. This is my testng file
![image](https://github.com/Test-Pro-Bootcamp/A45-QA-Automation-Framework/assets/130935020/b624f15c-e51f-49a2-b0c2-92908a14fad7)

https://github.com/Test-Pro-Bootcamp/A45-QA-Automation-Framework/blob/99b38aba4bafe01886c7bb451af0d7969ca8899f/src/test/resources/TestNG.xml

Berichicko commented:
> @Berichicko I think you meant to post this on someone else's feedback. This is my testng file ![image](https://user-images.githubusercontent.com/130935020/238392154-b624f15c-e51f-49a2-b0c2-92908a14fad7.png)
> 
> https://github.com/Test-Pro-Bootcamp/A45-QA-Automation-Framework/blob/99b38aba4bafe01886c7bb451af0d7969ca8899f/src/test/resources/TestNG.xml

Hello

Please, re-submit homework 25 for review

Berichicko commented:
> @Berichicko I think you meant to post this on someone else's feedback. This is my testng file ![image](https://user-images.githubusercontent.com/130935020/238392154-b624f15c-e51f-49a2-b0c2-92908a14fad7.png)
> 
> https://github.com/Test-Pro-Bootcamp/A45-QA-Automation-Framework/blob/99b38aba4bafe01886c7bb451af0d7969ca8899f/src/test/resources/TestNG.xml



```
            <class name="Pages.BasePage"/>
            <class name="Pages.LoginPage"/>
            <class name="Pages.HomePage"/>
```

![image](https://github.com/Test-Pro-Bootcamp/A45-QA-Automation-Framework/assets/104761482/82df5fb5-43d7-4d9b-9e38-c51c5ba8bc8f)


These are not tests, but descriptions of pages, web elements

**HINT**

Please add more test classes in TestNG

For example

![image](https://github.com/Test-Pro-Bootcamp/A45-QA-Automation-Framework/assets/104761482/459576da-4392-41b3-9631-50da15621c75)

rogarciap8 committed:
Homework25 UPDATED

Berichicko commented:
Please finish the homework!You got this problem, please correct it. 

![image](https://github.com/Test-Pro-Bootcamp/A45-QA-Automation-Framework/assets/104761482/bda5f8a3-1cb9-4320-ac13-844e818a92da)

![image](https://github.com/Test-Pro-Bootcamp/A45-QA-Automation-Framework/assets/104761482/886eb71e-8682-4adb-9380-7c0f70a143d2)

HINT

I'll try to help you, here's an example of parallel testing in 4 threads on a local machine:



_Download_

1) Code with Russian comments https://drive.google.com/file/d/1dQjnroe0pIiIqll_dT8rTwgAO5cekj4W/view?usp=share_link


2) Code with English comments https://drive.google.com/file/d/1H6MUQcisY_XQ2kn47AhrGU5TBVxli1ae/view?usp=share_link


**To run the tests, open the README file in IntelliJ IDEA**


![image](https://github.com/Test-Pro-Bootcamp/A45-QA-Automation-Framework/assets/104761482/3c7532eb-65f0-4854-823b-be41ba43622a)

**click**  ![image](https://github.com/Test-Pro-Bootcamp/A45-QA-Automation-Framework/assets/104761482/7313f1bd-6aba-4a26-8f83-dd510a432c8c)

_PS:  To unzip a ZIP file on a Mac_ https://www.winzip.com/en/learn/file-formats/zip/mac/

rogarciap8 committed:
HW 25 changes

rogarciap8 committed:
HW 25 Update

rogarciap8 committed:
HW25 TS

rogarciap8 committed:
Hw 25 Working

Changed default browser to Firefox. There is an issue with Chrome and the 4 threads. Works fine with other browsers.

Berichicko commented:
Hello 
Your homework 25 is checked
**You’re doing great! Keep it up! Very fine!** 
💖Best wishes, the team HW (Alex,  aliaksandr@testpro.io)!💖

![804deb64dc406a20a75da05d91a1a69e](https://user-images.githubusercontent.com/104761482/235961479-9afb2841-b657-46c9-99f8-2402f933b843.jpg)
